File: morkato/objects/types/player.py
from typing import (
  Literal,
  TypedDict,
  Union
)

PlayerBreed = Literal['HUMAN', 'ONI', 'HYBRID']

class Player(TypedDict):
  name:        str
  credibility: int
  
  guild_id: str
  id:       str
  breed:    PlayerBreed

  cash: int

  life:   int
  blood:  int
  breath: int
  exp:    int

  appearance:   Union[str, None]
  webhook_link: str

File: api/utils/player.ts
import type { Player as PrismaPlayer } from '@prisma/client'
import type { Player } from 'type:models/player'

export function format({
  name,
  surname,
  breed,
  cash,
  exp,
  credibility,
  history,
  guild_id,
  id,
  life,
  breath,
  blood,
  force,
  resistance,
  velocity,
  appearance,
  banner,
  updated_at
}: PrismaPlayer): Player {
  return {
    name: name,
    surname: surname,
    breed: breed,
    cash: cash,
    exp: exp,
    credibility: credibility,
    history: history,
    guild_id: guild_id,
    id: id,
    life: life,
    breath: breath,
    blood: blood,
    force: force,
    resistance: resistance,
    velocity: velocity,
    appearance: appearance,
    banner: banner,
    updated_at: updated_at === null ? null : updated_at.getTime()
  }
}